File: Backend/BloodBand.Models/OrganizationModel.cs
namespace BloodBand.Models
{
    public class OrganizationModel
    {
        public string? Name { get; set; }
        public string? Description { get; set; }

        public int? CountryId { get; set; }
        public int? StateId { get; set; }
        public int? DistrictId { get; set; }

        public string? Place { get; set; }
        public string? Pincode { get; set; }

        public string? PhoneNumber { get; set; }
        public string? Email { get; set; }

        public string? RegistrationNumber { get; set; }
        public string? RegistrationType { get; set; }
        public DateTime? RegistrationDate { get; set; }

        public string? LicenseNumber { get; set; }
        public string? LicenseIssuedBy { get; set; }
        public string? Password { get; set; }
    }
}

File: UI/bloodband-app/src/app/models/organization.model.ts
/**
 * Strong-typed interface matching the backend OrganizationModel data structure
 */
export interface OrganizationModel {
  name: string;
  description?: string;
  
  // Geographic Alignment Parameters
  countryId: number;
  stateId: number;
  districtId: number;
  place: string;
  pincode: string;

  // Regulatory Coordinates
  registrationNumber: string;
  registrationType: string;
  registrationDate: Date | string;
  licenseNumber: string;
  licenseIssuedBy: string;

  // Access Credentials & Contact Channels
  phoneNumber: string;
  email: string;
  password?: string;
}